app/src/androidTest/java/ch/hikemate/app/navigation/HikeMateAppNavigationTest.kt

```diff
@@ -1,67 +1,112 @@
+import androidx.compose.ui.test.performTextInput
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import ch.hikemate.app.HikeMateApp
 import ch.hikemate.app.ui.auth.SignInScreen
+import ch.hikemate.app.ui.auth.SignInWithEmailScreen
+import ch.hikemate.app.ui.components.BackButton
 import ch.hikemate.app.ui.navigation.Route
 import ch.hikemate.app.ui.navigation.Screen
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_ITEM_PREFIX
 import ch.hikemate.app.ui.navigation.TEST_TAG_SIDEBAR_BUTTON
 import ch.hikemate.app.ui.saved.TEST_TAG_SAVED_HIKES_SECTION_CONTAINER
-import ch.hikemate.app.ui.theme.HikeMateTheme
+import com.google.firebase.FirebaseApp
+import com.google.firebase.auth.FirebaseAuth
+import org.junit.Before
 import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
 
-// @RunWith(AndroidJUnit4::class)
+@RunWith(AndroidJUnit4::class)
 class HikeMateAppNavigationTest {
   // Set up the Compose test rule
   @get:Rule val composeTestRule = createComposeRule()
+  private val auth = FirebaseAuth.getInstance()
+  private val email = "navTesting@gmail.com"
+  private val password = "password"
 
-  // @Test
-  fun testInitialScreenIsAuthScreen() {
-    composeTestRule.setContent { HikeMateTheme { HikeMateApp() } }
+  @Before
+  fun setUp() {
+    composeTestRule.setContent { HikeMateApp() }
+    auth.signOut()
+  }
 
-    composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
+  private fun setupUser() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    FirebaseApp.initializeApp(context)
+
+    auth.createUserWithEmailAndPassword(email, password)
+    auth.signOut()
   }
 
-  // @Test
-  fun testNavigationToMapScreen() {
-    composeTestRule.setContent { HikeMateApp() }
+  private fun deleteUser() {
+    auth.currentUser?.delete()
+    auth.signOut()
+  }
 
+  @Test
+  fun testNavigationThroughAuthScreens() {
     composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
 
-    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).performClick()
-    composeTestRule.onNodeWithTag(Screen.MAP).assertIsDisplayed()
+    // Navigate to the sign in screen
+    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_EMAIL).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_GO_TO_SIGN_UP_BUTTON)
+        .performClick()
+    composeTestRule.onNodeWithTag(Screen.CREATE_ACCOUNT).assertIsDisplayed()
+
+    // Navigate back to the sign in screen
+    composeTestRule.onNodeWithTag(BackButton.BACK_BUTTON_TEST_TAG).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(BackButton.BACK_BUTTON_TEST_TAG).performClick()
+    composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
```

Comment:
Praise: Nice test! It really tests the navigation and checks that the back button goes only back one screen!